controllers/snapshot_controller.go

```diff
@@ -140,3 +163,26 @@ func (r *SnapshotReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 
 	return ctrl.Result{}, nil
 }
+
+// reconcileDeleteSnapshot removes the cached data that the snapshot was associated with if it exists.
+func (r *SnapshotReconciler) reconcileDeleteSnapshot(ctx context.Context, obj *deliveryv1alpha1.Snapshot) error {
+	patchHelper, err := patch.NewHelper(obj, r.Client)
+	if err != nil {
+		return fmt.Errorf("failed to reconcile delete: %w", err)
+	}
+
+	name, err := ocm.ConstructRepositoryName(obj.Spec.Identity)
+	if err != nil {
+		return fmt.Errorf("failed to construct name: %w", err)
+	}
+
+	if err := r.Cache.DeleteData(ctx, name, obj.Spec.Tag); err != nil {
+		if !strings.Contains(err.Error(), "MANIFEST_UNKNOWN") {
```

Comment:
Does this not return status code we could use instead of the string?

Reply:
It's a bit ugly, because the manifest unknown will be in the diagnostics list. Pf.

Reply:
Done. Verifying now if Unknown Manifest is in the list of diagnostic errors. I can't do better. :D